### PR TITLE
feat: deploy devnet on kubernetes

### PR DESCRIPTION
**Pull request by tjwelde**
## fixes KILTProtocol/ticket#738
This PR deploys the devnet on kubernetes

## How to test:
- Check, that after doing a change, the devnet is redeployed successfully 

## Checklist:

- [x] I have verified that the code works
- [x] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [x] I have [left the code in a better state](https://deviq.com/boy-scout-rule/)
- [x] I have documented the changes (where applicable)

### CODE REVIEW

File: .github/workflows/kube-dev.yml [hunk 1]
```diff
@@ -0,0 +1,109 @@
+name: Deploy to AWS EKS
+
+on:
+  push:
+    branches: 
+      - tw-738-kubernetes
```
Comment (weichweich) [inline .github/workflows/kube-dev.yml:6]: this probably needs to be changed before merging

File: .github/workflows/kube-dev.yml [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        docker push $ECR_REGISTRY/$ECR_REPOSITORY:$ECR_IMAGE_TAG
+        docker push $ECR_REGISTRY/$ECR_REPOSITORY:$SHA_IMAGE_TAG
+        echo "::set-output name=image::$ECR_REGISTRY/$ECR_REPOSITORY:$ECR_IMAGE_TAG"
+
+    # Here, we're creating the parent directory and writing out our decoded
+    # kubeconfig to the location we stated above.
+    - name: Configure Kubernetes
+      run: |
+        mkdir -p '${{ github.workspace }}/.kube' \
+          && echo '${{ secrets.KUBE_CONFIG}}' | base64 -d > $KUBECONFIG
```
Comment (weichweich) [inline .github/workflows/kube-dev.yml:63]: oh crazy! 😁 You have the whole config inside a secret!?

File: .github/workflows/kube-dev.yml [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+    - name: Set new image
+      env:
+        ECR_REGISTRY: ${{ steps.login-ecr.outputs.registry }}
+      run: |
+        kubectl -n devnet set image deployment/node-alice-deployment node-container=$ECR_REGISTRY/$ECR_REPOSITORY:$SHA_IMAGE_TAG
+        kubectl -n devnet set image deployment/node-bob-deployment node-container=$ECR_REGISTRY/$ECR_REPOSITORY:$SHA_IMAGE_TAG
+        kubectl -n devnet set image deployment/node-charlie-deployment node-container=$ECR_REGISTRY/$ECR_REPOSITORY:$SHA_IMAGE_TAG
+        kubectl -n devnet set image statefulset/node-full-deployment node-container=$ECR_REGISTRY/$ECR_REPOSITORY:$SHA_IMAGE_TAG
```
Comment (weichweich) [inline .github/workflows/kube-dev.yml:72]: We just exchange the binary without doing a runtime upgrade? Not sure if this works. I will talk to @Dudleyneedham and ask what he tested

File: .github/workflows/kube-dev.yml (hunk 2 above)
Comment (tjwelde) [reply .github/workflows/kube-dev.yml:63]: I was also surprised, that this is how you do it. 😅

File: .github/workflows/kube-dev.yml (hunk 3 above)
Comment (tjwelde) [reply .github/workflows/kube-dev.yml:90]: @Dudleyneedham tried to update the native binaries, before applying his substrate 2 runtime upgrade and it worked

**Review by weichweich — APPROVED** [review]
LGTM

**Pull request merged**